### ISSUE TITLE
Bash error when hitting tab after a required command

### ISSUE THREAD

**Issue by dtmistry** [spf13/cobra#407]
I've a root command without flags and a child command with some flags. Only one of the flag is required for the child command. 

The generated completion script is installed on my Mac using homebrew. 

Autocomplete works fine up to the required flag, but fails with the below error when I hit tab after the flag (for autocompleting a file location for a flag)

```
$ clutch add-spec [tab][tab]   
--ctxRoot=    -c
```

`
$ clutch add-spec -c /test -d ~/workspace[tab]
`

`
$ clutch add-spec -c /test -d ~/workspace/-bash: flaghash[${flagname}]: bad array subscript
`

`
$ clutch add-spec --ctxRoot=/test -d ~/workspace[tab]
`

`
$ clutch add-spec --ctxRoot=/test -d ~/work-bash: --ctxRoot=: attempted assignment to non-variable (error token is "=")
`

The error persist with whatever order of flags.

**Comment by johnmccabe** [comment]
This seems to be related to the version of bash on OSX and its handling of hashes, this fails for me:
```
./my-cli --gateway=jh bash: --gateway=: attempted assignment to non-variable (error token is "=")
```
Which is falling over here in the `__handle_flag()` method:
```
flaghash[${flagname}]=${flagvalue}
```
This fails with the default bash `GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin16)`, but works if I use brew to install a more recent version `GNU bash, version 4.4.12(1)-release (x86_64-apple-darwin16.3.0)`.

**Comment by johnmccabe** [comment]
@dtmistry I've raised a PR with a potential fix if you want to try that out.

**Comment by bacongobbler** [comment]
looks like this can be closed now that #520 has been merged. :)
